### DISCUSSION TITLE
Back to Back msgpack serialization to/from asio stream...

### DISCUSSION THREAD

**Discussion by byronfa** [nlohmann/json#2401] [Q&A] [answered]
Perhaps a simple question in search of code example(s) to look at.

I'm trying to parse a stream of MsgPack encoded json messages.  I was hoping naively that I could directly or indirectly pass a asio binary stream to json::from_msgpack() and loop on decoded messages.   Should that be possible or am I misunderstanding interfaces?

Most examples I see involve parsing a text string using json::parse, or when raw msgpack is involved, looping on partial buffer reads from the port.  

Thanks...

**Comment by nlohmann** [comment, marked as answer]
I am not exactly sure what you want to achieve. [`from_msgpack`](https://nlohmann.github.io/json/api/basic_json/from_msgpack/) has several overloads to read data from, most notably any pair of input iterators. If your data can be presented in such form, then it should work.

**Reply by nlohmann** [reply]
Do you need further assistance with this issue?
